### PR TITLE
upgrade: Add support for api/upgrade/mode API

### PR DESCRIPTION
**Pull request by rhafer**

### CODE REVIEW

**Comment by rhafer** [comment]
[(3) S56P4 API for forcing disruptive upgrade mode](https://trello.com/c/XY1ZBnMg/319-3-s56p4-api-for-forcing-disruptive-upgrade-mode)

File: lib/crowbar/client/app/upgrade.rb [hunk 1]
```diff
@@ -367,6 +367,24 @@ def database(mode)
           catch_errors(e)
         end
 
+        desc "mode UPGRADEMODE",
+          "Set/Get the upgrade mode (normal|non_disruptive)"
+
+        long_desc <<-LONGDESC
+          `mode` allows you to switch the upgrade mode between `non_disruptive` and `normal`.
+          Calling it without any arguments will return the currently selected upgrade mode.
```
Comment (jsuchome) [inline lib/crowbar/client/app/upgrade.rb]: It may be useful to write about the time when this can (or more likely cannot) be called.

Comment (rhafer) [inline lib/crowbar/client/app/upgrade.rb]: Added.

File: spec/crowbar/client/command/upgrade/mode_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#
+# Copyright 2016, SUSE Linux GmbH
```
Comment (rsalevsky) [inline spec/crowbar/client/command/upgrade/mode_spec.rb]: 2017

File: lib/crowbar/client/command/upgrade/mode.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            @request ||= Request::Upgrade::Mode.new(
+              args
+            )
+          end
+
+          def execute
+            unless ["normal", "non_disruptive", nil].include? args.mode
+              err "Invalid upgrade mode '#{args.mode}'. " \
+                  "Only 'normal' and 'non_disruptive' are valid upgrade modes"
+            end
```
Comment (rsalevsky) [inline lib/crowbar/client/command/upgrade/mode.rb]: Can you pls add an empty line after the end.

File: lib/crowbar/client/command/upgrade/mode.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#
+# Copyright 2015, SUSE Linux GmbH
```
Comment (rsalevsky) [inline lib/crowbar/client/command/upgrade/mode.rb]: 2017

File: spec/crowbar/client/command/upgrade/mode_spec.rb (hunk 2 above)
Comment (rhafer) [inline spec/crowbar/client/command/upgrade/mode_spec.rb]: done

File: lib/crowbar/client/command/upgrade/mode.rb (hunk 4 above)
Comment (rhafer) [inline lib/crowbar/client/command/upgrade/mode.rb]: done

File: lib/crowbar/client/command/upgrade/mode.rb (hunk 3 above)
Comment (rhafer) [inline lib/crowbar/client/command/upgrade/mode.rb]: done

**Pull request merged**